lib/collector/remote-method.js

```diff
@@ -107,17 +107,22 @@ RemoteMethod.prototype._post = function _post(data, nrHeaders, callback) {
   }
 
   if (options.compressed) {
-    var useGzip = this._config.compressed_content_encoding === 'gzip'
-    var compressor = useGzip ? zlib.gzip : zlib.deflate
-    compressor(data, function onCompress(err, compressed) {
-      if (err) {
-        logger.warn(err, 'Error compressing JSON for delivery. Not sending.')
-        return callback(err)
-      }
-
-      options.body = compressed
-      makeRequest()
-    })
+    try {
+      var useGzip = this._config.compressed_content_encoding === 'gzip'
+      var compressor = useGzip ? zlib.gzip : zlib.deflate
+      compressor(data, function onCompress(err, compressed) {
+        if (err) {
+          logger.warn(err, 'Error compressing JSON for delivery. Not sending.')
+          return callback(err)
+        }
+
+        options.body = compressed
+        makeRequest()
+      })
+    } catch (err) {
+      logger.warn(err, 'Error compressing JSON for delivery. Not sending.')
```

Comment:
It appears that if makeRequest() throws an exception, the logs will say that the error was in compressing, but could it be that the compression worked find and it was an error in making the request?